src/Bridge/Doctrine/Orm/ItemDataProvider.php

```diff
@@ -94,13 +102,25 @@ public function getItem(string $resourceClass, $id, string $operationName = null
 
         $repository = $manager->getRepository($resourceClass);
         $queryBuilder = $repository->createQueryBuilder('o');
+        $i = 0;
+        $identifiersObj = new \ArrayObject($identifiers);
+        $identifiersObjIterator = $identifiersObj->getIterator();
+
+        while ($identifiersObjIterator->valid()) {
+            $placeholder = 'id_'.$identifiersObjIterator->key();
+            $whereExpr[$i] = $queryBuilder->expr()->eq('o.'.$identifiersObjIterator->key(), ':'.$placeholder);
+
+            if ($i <= 0) {
+                $queryBuilder
+                    ->where($whereExpr[$i]);
+            } else {
+                $queryBuilder
+                    ->andWhere($whereExpr[$i]);
+            }
+            $queryBuilder->setParameter($placeholder, $identifiersObjIterator->current());
 
-        foreach ($identifiers as $propertyName => $value) {
-            $placeholder = 'id_'.$propertyName;
-
-            $queryBuilder
-                ->where($queryBuilder->expr()->eq('o.'.$propertyName, ':'.$placeholder))
-                ->setParameter($placeholder, $value);
+            ++$i;
+            $identifiersObjIterator->next();
         }
```

Comment:
The point of the iterator is to get rid of `$i`. You can easily have something more readable by putting that into a private function:

```php
$this->processIdentifiers($identifiers, $queryBuilder);

private function processIdentifiers(array $identifiers, QueryBuilder $queryBuilder)
{
    if (0 === count($identifiers)) {
        return;
    }
    $iterator = (new \ArrayObject($identifiers))->getIterator();

    $queryBuilder->where(
        $queryBuilder->expr()->eq(
            'o.'.$identifiersObjIterator->key(),
            ':id_'.$identifiersObjIterator->key()
        )
    );
    $iterator->next();

    while($iterator->valid()) {
        $queryBuilder->andWhere(
            $queryBuilder->expr()->eq(
                'o.'.$identifiersObjIterator->key(),
                ':id_'.$identifiersObjIterator->key()
            )
        );
        $iterator->next();
    }
}
```